python/mpc_cvxgen/mpc_cvxgen.py

```diff
@@ -0,0 +1,120 @@
+"""
+Author: Fangyu Wu, Edward Fang
+Email: fangyuwu@berkeley.edu, edward.fang@berkeley.edu
+
+The code is adapted from https://github.com/AtsushiSakai/PythonRobotics/tree/
+master/PathTracking/model_predictive_speed_and_steer_control.
+Its author is Atsushi Sakai.
+
+Reference Papers:
+- [Kinematic and Dynamic Vehicle Models for Autonomous Driving Control Design]
+(https://borrelli.me.berkeley.edu/pdfpub/IV_KinematicMPC_jason.pdf)
+"""
+import ctypes
+from ctypes import c_double, c_int
+from mpc_cvxgen.mpc import ModelPredictiveController
+import numpy as np
+import os
+
+path = os.path.abspath(__file__).split('/')[:-3]
+path = '{}/build/applysolver.so'.format('/'.join(path))
+if os.path.exists(path):
+    cdll = ctypes.CDLL(path)
+else:
+    raise ValueError('Shared library applysolver.so is not found.')
+c_double_p = ctypes.POINTER(c_double)
+cdll.apply_solver.argtypes = (
```

Comment:
I know I did this at first too, but it might be better to pass a pointer to a shared Python / C++ structure. I've updated my planning code: https://github.com/erdos-project/frenet_optimal_trajectory_planner/blob/master/FrenetOptimalTrajectory/fot_wrapper.py

There is a simpler working example here: https://stackoverflow.com/questions/42506296/converting-python-dictionary-to-ctypes-structure

A little tedious but greatly improves readability and potential for error in the code.